src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,75 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private RegistrationService registrationService = new RegistrationServiceImpl();
+    private User user;
+    private Storage storage;
+
+    @BeforeEach
+    public void setup() {
+        storage = new Storage();
+        user = new User();
```

Comment:
user should be fully setup here. When you in your test set password to null, and expect your exception to be thrown, it's actually thrown because login is null, and login check goes first. Your test don't actually test what you think they test

Reply:
Oleh's comment not fixed, you always put new empty user to test and always get same exception.
You need to fill user with all proper fields in BeforeEach method and then for each test change required field to perform proper testing (login to null, password too short etc.)